File: solutions/elixir/bird-count/1/lib/bird_count.ex
defmodule BirdCount do
  def today([]), do: nil

  def today(list) do
    List.first(list)
  end

  def increment_day_count(list) do
    case today(list) do
      nil -> [1]
      _ -> List.update_at(list, 0, &(&1 + 1))
    end
  end

  def has_day_without_birds?(list) do
    0 in list
  end

  def total(list) do
    Enum.reduce(list, 0, &(&1 + &2))
  end

  def busy_days(list) do
    Enum.count(list, &(&1 > 4))
  end
end

File: solutions/ruby/bird-count/1/bird_count.rb
class BirdCount
  def self.last_week
    [0, 2, 5, 3, 7, 8, 4]
  end

  def initialize(birds_per_day)
    @this_week_counts = birds_per_day
  end

  def yesterday
    @this_week_counts[-2]
  end

  def total
    @this_week_counts.sum
  end

  def busy_days
    visits = @this_week_counts.select { |x| x >= 5 }
    visits.count
  end

  def day_without_birds?
    @this_week_counts.any? 0
  end
end
